src/Components/Components/src/Binding/ModelBindingContext.cs

```diff
@@ -39,11 +40,161 @@ internal ModelBindingContext(string name, string bindingContextId, Predicate<Typ
     /// </summary>
     public string BindingContextId { get; }
 
+    /// <summary>
+    /// Retrieves the list of errors for a given model key.
+    /// </summary>
+    /// <param name="key">The key used to identify the specific part of the model.</param>
+    /// <returns>The list of errors associated with that part of the model if any.</returns>
+    public IReadOnlyList<FormattableString> GetErrors(string key) =>
+        _errors?.TryGetValue(key, out var bindingError) == true ? bindingError.ErrorMessages : Array.Empty<FormattableString>();
+
+    /// <summary>
+    /// Retrieves the list of errors for a given model key.
+    /// </summary>
+    /// <param name="key">The key used to identify the specific part of the model.</param>
+    /// <param name="formName">Form name for a form under this context.</param>
+    /// <returns>The list of errors associated with that part of the model if any.</returns>
+    public IReadOnlyList<FormattableString> GetErrors(string formName, string key) =>
+        _errorsByFormName?.TryGetValue(formName, out var formErrors) == true &&
+        formErrors.TryGetValue(key, out var bindingError) == true ? bindingError.ErrorMessages : Array.Empty<FormattableString>();
+
+    /// <summary>
+    /// Retrieves all the errors for the model.
+    /// </summary>
+    /// <returns>The list of errors associated with the model if any.</returns>
+    public IEnumerable<KeyValuePair<(object, string), IReadOnlyList<FormattableString>>> GetAllErrors()
```

Comment:
A bit of an unusually complex generic type to use in a public API. Fine for preview 6, but later on maybe we should look into whether we could use `FieldIdentifier` instead of the `(object, string)` part at least. I know that would involve moving some of this into `Microsoft.AspNetCore.Components.Forms` so I don't know for sure whether it's viable, but maybe it's something I could look at when working in this area in preview 7 if you don't mind.

Reply:
I've updated the API here to expose the BindingError directly, it's worth having the concept here.